Parsing stop-color with invalid values (i.e., "1234" - hashless color value)

rwlbuis opened the pull request:
#### dc180074984214d4813a306c36ecd0d3b23d3fca
<pre>
Parsing stop-color with invalid values (i.e., &quot;1234&quot; - hashless color value)
<a href="https://bugs.webkit.org/show_bug.cgi?id=265420">https://bugs.webkit.org/show_bug.cgi?id=265420</a>

Reviewed by Tim Nguyen.

Treat the hashless hex color quirk as invalid for stop-color since it is not
listed as one of the supported properties for it [1].

[1] <a href="https://quirks.spec.whatwg.org/#the-hashless-hex-color-quirk">https://quirks.spec.whatwg.org/#the-hashless-hex-color-quirk</a>

* LayoutTests/imported/w3c/web-platform-tests/svg/pservers/parsing/stop-color-invalid-expected.txt:
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
(WebCore::CSSParserFastPaths::maybeParseValue):

Canonical link: <a href="https://commits.webkit.org/300296@main">https://commits.webkit.org/300296@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/14ad1d0911e7ab5d6011c4b04e242a8941e4d54e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122059 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41761 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32431 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128622 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/74153 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123935 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42476 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50355 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92774 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61657 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/125011 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/33875 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109305 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73430 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32888 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27471 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72116 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103382 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27662 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131383 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48998 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37266 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101336 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49372 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105519 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101207 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25658 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46574 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24689 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45724 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48855 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54589 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48325 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51675 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/50005 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/460f6475e9e02f1723a4e7bf24069f0f7b45686d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51073

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/14ad1d0911e7ab5d6011c4b04e242a8941e4d54e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51073

rwlbuis committed:
Parsing stop-color with invalid values (i.e., "1234" - hashless color value)
https://bugs.webkit.org/show_bug.cgi?id=265420

Reviewed by Tim Nguyen.

Treat the hashless hex color quirk as invalid for stop-color since it is not
listed as one of the supported properties for it [1].

[1] https://quirks.spec.whatwg.org/#the-hashless-hex-color-quirk

* LayoutTests/imported/w3c/web-platform-tests/svg/pservers/parsing/stop-color-invalid-expected.txt:
* Source/WebCore/css/parser/CSSParserFastPaths.cpp:
(WebCore::CSSParserFastPaths::maybeParseValue):

Canonical link: https://commits.webkit.org/300296@main

tuankiet65 commented:
Thank you for fixing this 🙏
